sound/soc/sof/intel/mtl.c

```diff
@@ -255,9 +256,13 @@ static int mtl_dsp_pre_fw_run(struct snd_sof_dev *sdev)
 	if (ret < 0)
 		dev_err(sdev->dev, "failed to power up gated DSP domain\n");
 
-	/* make sure SoundWire is not power-gated */
-	snd_sof_dsp_update_bits(sdev, HDA_DSP_BAR, MTL_HFPWRCTL,
-				MTL_HfPWRCTL_WPIOXPG(1), MTL_HfPWRCTL_WPIOXPG(1));
+	hdev = sdev->pdata->hw_pdata;
+	/* Check scanned SoundWire capabilities exposed by DSDT */
+	if (hdev->info.handle && hdev->info.link_mask > 0)
+		/* make sure SoundWire is not power-gated */
+		snd_sof_dsp_update_bits(sdev, HDA_DSP_BAR, MTL_HFPWRCTL,
+					MTL_HfPWRCTL_WPIOXPG(1), MTL_HfPWRCTL_WPIOXPG(1));
+
```

Comment:
What problem are you trying to fix @yongzhi1 ?

Is this to save on power?

Reply:
Hi, @plbossart , yes, ACE_PGD7(IO1) is not power gated when I run I2S playback, with this experiment, IO1 is gated when testing with `aplay -Dhw:0,1 -c2 -r48000 -fS16_LE /dev/zero `  with NOCODEC tplg.

This is however, not an issue with idle (D3HOT) state, it just happen during playback.

Reply:
I guess similar issue could be seen on HDA only SKU, right?

Reply:
I do not have a HDA only setup to check, the logic works fine on I2S/SDW config so far.